connectors/sources/sharepoint_online.py

```diff
@@ -1352,43 +1354,12 @@ def _is_site_admin(user):
         access_control = set()
         site_admins_access_control = set()
 
-        async for user_information in self.client.user_information_list(site["id"]):
-            user = user_information["fields"]
-
+        async for role_assignment in self.client.site_role_assignments(site['webUrl']):
             user_access_control = set()
+            user_access_control.update(await self._get_access_control_from_role_assignment(role_assignment))
+            member = role_assignment["Member"]
 
-            if is_domain_group(user):
```

Comment:
probably need to check to make sure this didn't leave unused methods that can be cleaned up